src/inlineSuggestions/acceptInlineSuggestion.ts

```diff
@@ -36,9 +43,32 @@ export default async function acceptInlineSuggestion(
           editor,
           prefix
         ));
+
+    const vimActive = extensions.getExtension("vscodevim.vim")?.isActive;
+    if (vimActive) {
+      await vimReturnToInsertMode(currentSuggestion);
+    }
   }
 }
 
+async function vimReturnToInsertMode(suggestion: ResultEntry) {
+  await commands.executeCommand("extension.vim_escape");
+  await commands.executeCommand("extension.vim_insert");
+  const nextCursorPosition =
+    suggestion.new_prefix.length + suggestion.new_suffix.length;
+  setImmediate(() => {
+    moveCursorToPosition(nextCursorPosition);
+  });
+}
+
+function moveCursorToPosition(nextCursorPosition: number) {
```

Comment:
a better name imo can be `moveCursorRight(steps: number)`
or something that conveys the fact that this function moves the cursor to a relative position to the right